packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,396 @@
+
+      expect(checkbox).toBeInTheDocument();
+    });
+  });
+
+  describe('required variant', () => {
+    it('renders a design system BEM modifier class name', async () => {
+      const { container } = await render('<input [required]="required" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          required: true,
+        },
+      });
+      const checkbox = container.querySelector('input[type="checkbox"]:required');
+
+      expect(checkbox).toHaveClass('utrecht-checkbox--required');
+    });
+
+    it('is not required by default', async () => {
+      const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      fixture.detectChanges();
+      const checkbox = screen.getByRole('checkbox');
+
+      expect(checkbox).not.toBeRequired();
+    });
+
+    it('omits non-essential required attributes when not required', async () => {
+      await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      const checkbox = screen.getByRole('checkbox');
+
+      expect(checkbox).not.toHaveAttribute('aria-required');
+
+      expect(checkbox).not.toHaveAttribute('required');
+    });
+
+    it('can have a required state', async () => {
+      const { container } = await render('<input [required]="required" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          required: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]:required');
+
+      expect(checkbox).toBeRequired();
+    });
+
+    it('can have a required state in CSS', async () => {
+      const { container } = await render('<input [required]="required" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          required: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]:required');
+
+      expect(checkbox).toBeInTheDocument();
+    });
+  });
+
+  it('can be hidden', async () => {
+    const { container } = await render('<input [hidden]="[hidden]" type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+      componentProperties: {
+        hidden: true,
+      },
+    });
+
+    const checkbox = container.querySelector('input[type="checkbox"]');
+
+    expect(checkbox).not.toBeVisible();
+  });
+
+  it('can have a custom class name', async () => {
+    const { container } = await render('<input [class]="[class]" type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+      componentProperties: {
+        class: 'custom-class',
+      },
+    });
+
+    const checkbox = container.querySelector('input[type="checkbox"]');
+
+    expect(checkbox).toHaveClass('custom-class');
```

Comment:
Please update this to do a slightly different test: `'can have an additional class name'`. The also check that the BEM class name is still there: the component must combine class names, not overwrite the BEM class names.

Reply:
I'm gonna update the test text here, also I will add `expect(checkbox).toHaveClass('utrecht-checkbox--html-input utrecht-checkbox--custom');` addition check to make sure we still have the another classes, and here is the output from the `debug()` function

```
    <input
      class="custom-class utrecht-checkbox--html-input utrecht-checkbox--custom utrecht-checkbox"
      type="checkbox"
      utrecht-checkbox=""
    />
```html